src/components/web-form-renderer.vue

```diff
@@ -0,0 +1,304 @@
+<!--
+Copyright 2025 ODK Central Developers
+See the NOTICE file at the top-level directory of this distribution and at
+https://github.com/getodk/central-frontend/blob/master/NOTICE.
+
+This file is part of ODK Central. It is subject to the license terms in
+the LICENSE file found in the top-level directory of this distribution and at
+https://www.apache.org/licenses/LICENSE-2.0. No part of ODK Central,
+including this file, may be copied, modified, propagated, or distributed
+except according to the terms contained in the LICENSE file.
+-->
+
+<template>
+  <loading :state="initiallyLoading"/>
+  <template v-if="!initiallyLoading">
+    <OdkWebForm :form-xml="formVersionXml.data" :fetch-form-attachment="getAttachment" @submit="handleSubmit"/>
+  </template>
+
+  <modal v-bind="previewModal" hideable backdrop @hide="previewModal.hide()">
+    <template #title>{{ $t('WebformFill.previewModal.title') }}</template>
+    <template #body>
+      {{ $t('WebformFill.previewModal.body') }}
+      <div class="modal-actions">
+        <button type="button" class="btn btn-primary" @click="previewModal.hide()">
+          {{ $t('action.close') }}
+        </button>
+      </div>
+    </template>
+  </modal>
+
+    <modal v-bind="submissionModal" hideable backdrop @hide="submissionModal.hide()">
```

Comment:
this modal can be move to separate file, but I am expecting UI changes to this so I will move it along with those changes.

Reply:
I left a comment about this below, but I think it'd be nice to move it as part of the current PR if that's possible.

Reply:
Ah never mind, my other comment was about the new modal in `FormSettings` for toggling `webformsEnabled`. I wanted to move that modal into its own component because it's more complicated, e.g., it sends a request. The modal here is a different one though and is just static text, so I think it's OK to leave it where it is for now.